log4j-api/src/main/java/org/apache/logging/log4j/util/PropertiesUtil.java

```diff
@@ -649,45 +674,57 @@ public static Map<String, Properties> partitionOnCommonPrefixes(
      * @return true if system properties tell us we are running on Windows.
      */
     public boolean isOsWindows() {
-        return getStringProperty("os.name", "").startsWith("Windows");
+        return SystemPropertiesPropertySource.getSystemProperty("os.name", "").startsWith("Windows");
+    }
+
+    static Duration parseDuration(final String value) {
+        final Matcher matcher = DURATION_PATTERN.matcher(value);
+        if (matcher.matches()) {
+            return Duration.of(TimeUnit.parseAmount(matcher), TimeUnit.parseUnit(matcher));
+        }
+        throw new DateTimeParseException("Invalid duration value: should be in the format nnn[unit].", value, 0);
```

Comment:
I don't think someone who is not able to correctly type a duration value can understand what is expected from them by looking at the hinted `nnn[unit]` format. Can we make the example/pattern a little bit more human-friendly?